urllib3/util/retry.py

```diff
@@ -290,5 +290,31 @@ def __repr__(self):
                     cls=type(self), self=self)
 
 
+def handle_redirect(response, requester, retries, method, url, *args, **kwargs):
+    """
+    Abstraction of the redirect process so that we can access it from
+    multidomain levels in addition to from the :class:`.ConnectionPool`.
+    """
+    # urljoin will give us either a full URL or just a path, based on what
+    # the original URL is, making it appropriate for the ConnectionPool
+    # level as well as the PoolManager or SessionManager
+    redirect_location = urljoin(url, response.get_redirect_location())
+    if response.status == 303:
+        method = 'GET'
```

Comment:
This is somewhat incomplete. HEAD requests should also stay HEAD requests (a general rule in redirects). There are also questions about whether you should go further in emulating browsers, though that's up to you.